libs/mongodb/tests/integration_tests/test_retrievers.py

```diff
@@ -160,6 +161,18 @@ def test_fulltext_retriever(
         search_field=PAGE_CONTENT_FIELD,
     )
 
+    # Wait for the search index to complete.
+    search_content = dict(index=SEARCH_INDEX_NAME, wildcard=dict(query="*", path=PAGE_CONTENT_FIELD, allowAnalyzedField=True))
+    n_docs = collection.count_documents({})
+    t0 = time()
+    while True:
+        if (time() - t0) > TIMEOUT:
+            raise TimeoutError(f'Search index {SEARCH_INDEX_NAME} did not complete in {TIMEOUT}')
+        results = collection.aggregate([{ "$search": search_content }])
```

Comment:
This looks really good. The full-text equivalent to what we do with $vectorSearch. Nice.